### PR TITLE
MVP Changes from Coco

### PR DESCRIPTION
**Pull request by cocovsack**
Fixes aesthetic and functionality of display page by adding tag gallery, more spoonacular information, and scaling. Also forces users to log in before adding tags

### CODE REVIEW

File: src/main/webapp/style.css [hunk 1]
```diff
@@ -662,4 +736,55 @@ h1 {
...
+  margin-left: -120px;
+  margin-top: 70px;
+  padding-top: 0;
+  width: auto;
+}
+
+
+
+
+
```
Comment (joeychin01) [inline src/main/webapp/style.css:790]: nit: remove extra lines

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -451,26 +456,68 @@ function createRecipeElement(recipe, pictureWrap) {
...
+        const loginLink = document.createElement('a');
+        loginLink.className = "login-full-link upload";
+        loginLink.innerText = "LOGIN";
+        loginLink.href = userInfo.loginUrl;
+
+        document.body.append(loginPopup);
+        loginPopup.append(loginPopupText);
+        loginPopup.append(loginLink);
+        document.body.style.overflow = "hidden";
+        document.getElementById("display-recipes").style.opacity = "0.2";
```
Comment (liguanghua) [inline src/main/webapp/script.js:503]: Consider extracting these into a helper function for not-logged-in users.

Also extract a helper function for the else-branch

File: src/main/webapp/style.css [hunk 3]
```diff
@@ -168,24 +173,24 @@
...
+/* .recipe-tag:after, .recipe-tag:before {
   border: solid transparent;
   content: " ";
   height: 0;
   position: absolute;
   right: 100%;
   top: 50%;
   width: 0;
-}
+} */
```
Comment (linxiz) [inline src/main/webapp/style.css:193]: remove if not used. You could check history for previous version.

File: src/main/webapp/script.js [hunk 4]
```diff
@@ -587,13 +634,48 @@ function createAlertElement(iconName, innerText) {
   return clone;
 }
 
+function createElementFromHTML(htmlString) {
+  var div = document.createElement('div');
+  div.innerHTML = htmlString.trim();
+
+  // Change this to div.childNodes to support multiple top-level nodes
```
Comment (linxiz) [inline src/main/webapp/script.js:641]: nit: is this a TODO? If so, add "TODO: " at the begining

**Pull request merged**